cmd/vela-worker/operate.go

```diff
@@ -118,6 +119,12 @@ func (w *Worker) operate(ctx context.Context) error {
 	//nolint:contextcheck // ignore passing context
 	w.Queue, err = queue.New(w.Config.Queue)
 	if err != nil {
+		registryWorker.SetStatus(constants.WorkerStatusError)
+		_, res, ers := w.VelaClient.Worker.Update(registryWorker.GetHostname(), registryWorker)
+		if ers != nil {
```

Comment:
i think we also need to check to make sure res is not nil somewhere, since we grab the statuscode from it for the error